src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,175 @@
+
+    def __init__(
+        self,
+        optimization: Optional[Literal["trace", "script", "onnx", "tensorrt"]] = None,
+        server: Literal["fastapi", "ml_server", "torchserve", "sagemaker"] = "fastapi",
+        host: str = "127.0.0.1",
+        port: int = 8080,
+        timeout: int = 10,
+        exit_on_failure: bool = True,
+    ):
+        super().__init__()
+        fastapi_installed = _RequirementAvailable("fastapi")
+        if not fastapi_installed:
+            raise ModuleNotFoundError(fastapi_installed.message)
+        uvicorn_installed = _RequirementAvailable("uvicorn")
+        if not uvicorn_installed:
+            raise ModuleNotFoundError(uvicorn_installed.message)
+
+        # TODO: Add support for the other options
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+        # TODO: Add support for testing with those server services
+        if server != "fastapi":
+            raise NotImplementedError("Only the fastapi server is currently supported.")
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.timeout = timeout
+        self.exit_on_failure = exit_on_failure
+        self.resp: Optional[requests.Response] = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningModule") -> None:
+        if isinstance(trainer.strategy, _NOT_SUPPORTED_STRATEGIES):
+            raise Exception(
+                f"The current strategy {trainer.strategy} used "
+                "by the trainer isn't supported for sanity serving yet."
+            )
+
+        if not isinstance(servable_module, ServableModule):
+            raise TypeError(f"The provided model should be subclass of {ServableModule.__qualname__}.")
+
+        if not is_overridden("configure_payload", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_payload` method needs to be overridden.")
+        if not is_overridden("configure_serialization", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_serialization` method needs to be overridden.")
+        if not is_overridden("serve_step", servable_module, ServableModule):
+            raise NotImplementedError("The `serve_step` method needs to be overridden.")
+
+        # Note: The Trainer needs to be detached from the pl_module before starting the process.
+        # This would fail during the deepcopy with DDP.
+        servable_module.trainer = None
+
+        process = Process(target=self._start_server, args=(servable_module, self.host, self.port, self.optimization))
+        process.start()
+
+        servable_module.trainer = trainer
+
+        ready = False
+        t0 = time.time()
+        while not ready:
+            try:
+                resp = requests.get(f"http://{self.host}:{self.port}/ping")
```

Comment:
Would it make sense to test that the model can be served with specific inference servers, like Triton or MLServer? This would ensure that the environment is as close to prod as similar. Conscious that this would mean having separate validators for each serving engine though (i.e. one for Sagemaker, one for the inference servers listed above, one for Seldon Core, etc.)....